galaxyproject/ansible-galaxy
# Travis-CI

A basic Travis-CI implementation which tests `ansible-galaxy`'s build process.

There are a few things left to do such as test conditional tasks and to write tests of the running build.
 
Using `root` as the Ansible `remote_user` fails with Travis (https://github.com/travis-ci/travis-ci/issues/1091). Therefore, become was added to this PR. The Ansible `become` statement allows a sudo user who is not `root` and does not have to be `{{ galaxy_user }}` to run the playbook. Tasks were also added to fix file permissions where needed.

The `become` half of this PR adds a security benefit (`root` user access is not needed over SSH and the `galaxy_user` who owns the Galaxy files does not need sudo) and makes sure that resulting files are owned by the `galaxy_user` instead of the `remote_user`.

To do before merge:
- test that the server can be started
- check that all files are owned by `{{ galaxy_user }}`
  - done manually
  - redirecting standard output to error does not fail travis 👎 
- update `README.md` playbook example
  - In the advanced example `s/become_user/galaxy_user/g` & test
- test conditional tasks (e.g., download.yml)
  - semi-dependent on other PRs
- check that the DB creation python scripts can be initially ran with sudo instead of as the `galaxy_user`
  - After running this PR on a VM, `sh run.sh` works but there may be issues in the DB still..?

Some of the commits include more info.

`.travis.yml` follows [ansible-galaxy-extras](https://github.com/galaxyproject/ansible-galaxy-extras/blob/master/.travis.yml) & [ansible-galaxy-tools](https://github.com/galaxyproject/ansible-galaxy-tools/blob/master/.travis.yml) as templates.

Thanks @drosofff and @eschen42!
